develop/tutorials/articles/150-workflow/01-creating-workflow-definitions/01-tasks.markdown

```diff
@@ -0,0 +1,275 @@
+
+To find all the resource actions that have been configured in @product@, you
+need access to the Roles Admin application in the Control Panel (in other words,
+you need permission for the VIEW action on the roles resource).
+
+- Navigate to Control Panel &rarr; Users &rarr; Roles.
+- Add a new Regular Role. See the 
+  [article on managing roles](/discover/portal/-/knowledge_base/7-0/roles-and-permissions)
+  for more information.
+- Once the role is added, navigate to the Define Permissions interface for the
+  role.
+- Find the resource whose action you want to use for defining your workflow
+  assignment.
+
+Here's what the assignment's XML looks like:
+
+    <assignments>
+        <resource-actions>
+            <resource-action>UPDATE</resource-action>
+        </resource-actions>
+    </assignments>
+
+Now when the workflow proceeds to the task with the resource action assignment,
+users with `UPDATE` permission on the resource (for example, Message Boards
+Messages) is notified of the task and can assign it to themselves (if the
+notification is set to Task Assignees). Specifically, users see the tasks
+in their *My Workflow Tasks* application under the tab *Assigned to My Roles*.
+
+Use all upper case letters for resource action names. Here are some common
+resource actions:
+
+    UPDATE
+    ADD
+    DELETE
+    VIEW
+    PERMISSIONS
+    SUBSCRIBE
+    ADD_DISCUSSION
+
+Determine the probable resource action name from the permissions screen for a
+resource. For example, in Message Boards, one of the permissions displayed on
+that screen is *Add Discussion*. Convert that to all uppercase and replace the
+space with an underscore, and you have the action name. 
+
+## Task Timers
+
+Task timers trigger an action after a specified time period passes. Timers are
+useful for ensuring a task does not go unattended for a long time. Available
+timer actions include sending an additional notification, reassigning the asset,
+or creating a timer action.
+
+    <task-timers>
+        <task-timer>
+            <name></name>
+            <delay>
+                <duration>1</duration>
+                <scale>hour</scale>
+            </delay>
+            <blocking>false</blocking>
+            <timer-actions>
+                <timer-notification>
+                    <name></name>
+                    <template></template>
+                    <template-language>text</template-language>
+                    <notification-type>im</notification-type>
+                </timer-notification>
+            </timer-actions>
+        </task-timer>
+    </task-timers>
+
+The above task timer creates a notification. Specify a time period in the
+`<delay>` tag, and specify what action to take after the delay is up in the
+`<timer-actions>` block. The `<blocking>` element is used to specify whether the
+timer actions may recur. If `<blocking>false</blocking>`, make the timer action
+recur using the `<recurrence>` tag:
+
+    <recurrence>
```

Comment:
The **recurrence** tag should be nested to the **task-timer** tag? I was unsure about that during the reading. 😞

Reply:
Yeah, it's nested below `<task-timer>`, at the same level as `<delay>`. I'll make this more clear.

Reply:
Nice! Thanks. 😄